Fix bug in 16-bit frame length when buffer is a subarray

jawj opened the pull request:
I was just bitten by this issue.

Because `buffer` could be a [subarray](https://nodejs.org/api/buffer.html#bufsubarraystart-end), the `byteOffset` and `byteLength` arguments to `new DataView()` are required, otherwise the `bodyLength` value may be written at the wrong offset.

Alternatively, it looks like it would be simpler (and I've tested it works fine) to replace the patched line with `buffer.writeUInt16BE(bodyLength, 2)`.

mcollina reviewed (commented):
Thanks for opening a PR! Can you please add a unit test?

jawj commented:
I wasn't exactly sure what you kind of test you wanted, but I've added a simple new test for `WebsocketFrameSend` that (almost certainly) fails without my fix.

I also switched to using `writeUInt16BE` instead of correcting the `DataView` creation, because it's simpler and more efficient and I couldn't see any reason why not. :)

It would be much appreciated if this fix could be released ASAP: a project I'm working on is blocked on this.

mcollina reviewed (approved):
lgtm

jawj commented:
Fab! Do you know how soon this will make it into the npm package?

jawj commented:
@mcollina Sorry to chase, but very keen to see this released to npm soon, because WebSockets are badly broken in the meantime. Any idea when that might happen?
